### ISSUE TITLE
Empty relationship data is represented as empty array instead as empty object

### ISSUE THREAD

**Issue by bartje321** [neomerx/json-api#217]
Code in schema:
```
public function getRelationships($resource, $isPrimary, array $includeRelationships)
{
    return [
        'foo' => []
    ];
}
```
Expected:
```
...
relationships:{
    foo:{}
}
...
```
Actual:
```
...
relationships:{
    foo:[]
}
...
```
Possible sollution:
[src/Document/Presenters/ElementPresenter.php](https://github.com/neomerx/json-api/blob/master/src/Document/Presenters/ElementPresenter.php#L111)
Change
```
$target[$parentType][$parentId][Document::KEYWORD_RELATIONSHIPS][$name] = $representation;
```
To
```
$target[$parentType][$parentId][Document::KEYWORD_RELATIONSHIPS][$name] = (object) $representation;
```

**Comment by lindyhopchris** [comment]
That is not compliant with the spec, which says:

> A “relationship object” MUST contain at least one of the following: [`links`, `data`, `meta`]
https://jsonapi.org/format/#document-resource-object-relationships

So neither an empty array or empty object is allowed.

**Comment by neomerx** [comment]
I think an extra check with `assert` should be added to validate data from `getRelationships`.

**Comment by neomerx** [comment]
An extra assert is added. You will see an error message similar to this

> A `people.foo` relationship must contain at least data, links or meta. in /.../json-api/src/Schema/BaseSchema.php:342

Where `people` would be your resource type.

**Comment by neomerx** [comment]
Released as 2.0.7
